src/cli.ts

```diff
@@ -25,7 +25,7 @@ program
 /**
 Arguments parsed for program.
 */
-interface IArgs extends commander.ICommand {
+interface IArgs extends program.ParseOptionsResult {
```

Comment:
@Glavin001 I'm not sure how/where this interface is to be used, since `program.parse` below returns a `command`

Reply:
The `command` type is generic. I want `IArgs` to provide an interface which is aware of what we expect the arguments to be parsed from `process.argv`.

Please add `IArgs` to `programArgs` below.

Reply:
Perhaps `ParseOptionsResult` is not the proper interface/type?  https://github.com/tj/commander.js/blob/master/typings/index.d.ts

Reply:
Yeah `ParseOptionsResult` looks good, since it has `args`.

```
interface ParseOptionsResult {
        args: string[];
        unknown: string[];
}
```

However, the important part is adding types for the other parsed arguments:

```
interface IArgs extends program.ParseOptionsResult {
  args: string[];
  language?: string;
  outFile?: string;
  replace?: boolean;
  configFile?: string;
  configJson?: string;
}
```

such that extracting them from `programArgs` works:

```
const {
  args: files,
  language,
  outFile,
  replace,
  configFile,
  configJson
} = programArgs;
```

and each of those variables has a type.

Obviously, there is no way for TypeScript and Commander to know the types of these command-line arguments, because they are evaluated at runtime by:

```
program
  .version(pkg.version)
  .usage("[options] [files...]")
  .option("-l, --language <language>", "Language of file to beautify")
  // .option("-i, --input-file [file]", "Input file to be beautified")
  .option("-o, --out-file <file>", "Output file of beautified results")
  .option("-r, --replace", "Replace file(s) with beautified results", false)
  .option("-c, --config-file <file>", "Beautifier configuration file")
  .option("--config-json <JSON>", "Beautifier configuration as stringified JSON")
```

So instead I manually added the types to `IArgs`.

Reply:
Makes sense.  But I'm thinking it actually should be extending `Command` since it also [has args](https://github.com/tj/commander.js/blob/master/typings/index.d.ts#L29), and that is what is returned from program.parse.